### ISSUE TITLE
OpenID returns invalid check_authentication request to LabWiki

### ISSUE THREAD

**Issue by tcmitchell** [GENI-NSF/geni-portal#869]
LabWiki is receiving "invalid check_authentication request" errors from the portal when using the OpenID authentication. The exact text of the error shown by LabWiki is:

  Authentication failed. Server https://portal.geni.net/server/server.php responds that the 'check_authentication' call is not valid

  Try again

This happens sometimes but not all the time. It happens frequently enough to be disturbing, especially for their upcoming tutorial. They estimate it happens between every third to every 10th time.

It's not clear whether the error is on the server side or the client side. The server side is not leaving anything in the logs that indicates a problem. It was easy to replicate the issue.

 _Imported from trac ticket [#869](http://trac.gpolab.bbn.com/proto-ch/ticket/869),  created by tmitchel on 10-23-2013 at 14:01, last modified: 05-16-2014 at 16:24_

**Comment by tcmitchell** [comment]
[Google](https://www.google.com/search?q=responds+that+the+%27check_authentication%27+call+is+not+valid) turns up some hits related to ruby-openid. The top 5 hits on google are:

 * https://github.com/openid/ruby-openid/issues/1
 * https://github.com/nbudin/devise_openid_authenticatable/issues/7
 * https://github.com/openid/ruby-openid/issues/51
 * http://developer.yahoo.com/forum/OpenID-General-Discussion/check-authentication-issue-with-ruby-openid/1264356406000-83688213-2436-39d0-85c0-ef1b75ff8ba3
 * http://stackoverflow.com/questions/1740138/openid-check-authentication-not-working

_Trac comment by tmitchel (github user: tcmitchell) on 10-23-2013 at 14:19_

**Comment by ahelsing** [comment]
One possibly interesting piece of information I stumbled upon today.  This issue seems to not occur for all users.
Jack reported this morning that he was not able to reproduce this problem.  After I tried this with both of my portal
users today, I found that I was able to reproduce it with user "johren" but not with "johren2".  I tried over 20 times in a row
with both.  I also swapped browsers and tried again.  The problem followed the user (johren).

_Trac comment by johren on 12-05-2013 at 19:23_
